rest/route/agent_test.go

```diff
@@ -728,3 +728,80 @@ func TestUpsertCheckRunParse(t *testing.T) {
 	assert.Equal(t, utility.FromStringPtr(r.checkRunOutput.Title), "This is my report")
 	assert.Equal(t, len(r.checkRunOutput.Annotations), 1)
 }
+
+func TestCreateGitHubDynamicAccessToken(t *testing.T) {
+	route := "/task/%s/github_dynamic_access_token/%s/%s"
+	owner := "owner"
+	repo := "repo"
+	taskID := "taskID"
+	json := []byte(`{ "checks": "read", "actions": "write" }`)
+
+	for tName, tCase := range map[string]func(ctx context.Context, t *testing.T, gh *createGitHubDynamicAccessToken, env *mock.Environment){
+		"ParseErrorsOnEmptyOwner": func(ctx context.Context, t *testing.T, handler *createGitHubDynamicAccessToken, env *mock.Environment) {
+			url := fmt.Sprintf(route, taskID, "", repo)
+			request, err := http.NewRequest(http.MethodGet, url, bytes.NewReader(json))
+			require.NoError(t, err)
+
+			options := map[string]string{"owner": "", "repo": repo, "taskID": taskID}
+			request = gimlet.SetURLVars(request, options)
+
+			assert.ErrorContains(t, handler.Parse(ctx, request), "missing owner")
+		},
+		"ParseErrorsOnEmptyRepo": func(ctx context.Context, t *testing.T, handler *createGitHubDynamicAccessToken, env *mock.Environment) {
+			url := fmt.Sprintf(route, taskID, owner, "")
+			request, err := http.NewRequest(http.MethodGet, url, bytes.NewReader(json))
+			require.NoError(t, err)
+
+			options := map[string]string{"owner": owner, "repo": "", "taskID": taskID}
+			request = gimlet.SetURLVars(request, options)
+
+			assert.ErrorContains(t, handler.Parse(ctx, request), "missing repo")
+		},
+		"ParseErrorsOnEmptyTaskID": func(ctx context.Context, t *testing.T, handler *createGitHubDynamicAccessToken, env *mock.Environment) {
+			url := fmt.Sprintf(route, "", owner, repo)
+			request, err := http.NewRequest(http.MethodGet, url, bytes.NewReader(json))
+			require.NoError(t, err)
+
+			options := map[string]string{"owner": owner, "repo": repo, "taskID": ""}
+			request = gimlet.SetURLVars(request, options)
+
+			assert.ErrorContains(t, handler.Parse(ctx, request), "missing taskID")
+		},
+		"ParseErrorsOnEmptyBody": func(ctx context.Context, t *testing.T, handler *createGitHubDynamicAccessToken, env *mock.Environment) {
+			url := fmt.Sprintf(route, taskID, owner, repo)
```

Comment:
Permissions will be an optional field, so we may want to only error if the body is not proper json or if it doesn't parse  to github permissions

Reply:
I'm down to make it optional but do you know of an example where we do that? I can't find one and I can only thing to do some weird manually error checking.

Reply:
Thinking about it more, it probably doesn't make sense to let users specify `no permission` in the command, because then it will do nothing. So if the user doesn't specify any permissions, we can technically pass in an empty struct to indicate all permissions, so I guess we don't have to ever use a nil body. Does that make sense?

Reply:
That makes sense! I'll leave it as is so that if it is empty, it should error.

Also side question, do users often call agent routes themselves?

Reply:
no, I meant making it optional on the command and the user not specifying permissions on the command